src/daemon/bitcoind/actions.rs

```diff
@@ -581,6 +606,35 @@ fn comprehensive_rescan(
                 }
             }
 
+            // Fourth layer: if the Cancel transaction *only* was confirmed and becomes unconfirmed,
+            // we just mark it as such. The bitcoind wallet will take care of the rebroadcast.
+            if matches!(vault.status, VaultStatus::Canceled) {
+                let cancel_tx = db_cancel_dbtx(&db_tx, vault.id)?.ok_or_else(|| {
+                    BitcoindError::Custom(format!(
+                        "Vault '{}' has status '{}' but no Cancel in database!",
+                        vault.deposit_outpoint, vault.status
+                    ))
+                })?;
```

Comment:
You could use the patch i made for the previous commit to fall back to generating it. Otherwise, logging as error instead of crashing would be better imo

Reply:
Pushed it myself